common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -139,10 +139,309 @@ static bool dwpal_get_channel_scan_freq(const std::vector<unsigned int> &channel
+        } break;
+
+        case ie_type::TYPE_BSS_LOAD: {
+            if (length != 5) {
+                LOG(ERROR) << "TYPE_BSS_LOAD doesn't match min and max length criteria";
+                break;
+            }
+            results.channel_utilization = (uint32_t)(data[2] / 255);
+        } break;
+
+        case ie_type::TYPE_RSN: {
+            if (length < 2) {
+                LOG(ERROR) << "TYPE_RSN doesn't match min and max length criteria";
+                break;
+            }
+            results.encryption_mode.push_back(eEncryption_Mode_AES);
+            results.security_mode_enabled.push_back(eSecurity_Mode_WPA2);
+        } break;
+
+        case ie_type::TYPE_EXTENDED_SUPPORTED_RATES: {
+
+            if (results.operating_frequency_band == eOperating_Freq_Band_2_4GHz) {
+                results.supported_standards.push_back(eStandard_802_11g);
+                results.operating_standards = eStandard_802_11g;
+            }
+
+            get_supprates(data, length, results);
+
+        } break;
+
+        case ie_type::TYPE_HT_OPERATION: {
+            if (length != 22) {
+                LOG(ERROR) << "TYPE_HT_OPERATION doesn't match min and max length criteria";
+                break;
+            }
+            get_ht_oper(data, results);
+        } break;
+
+        case ie_type::TYPE_VHT_OPERATION: {
+            if (length < 5) {
+                LOG(ERROR) << "TYPE_VHT_OPERATION doesn't match min and max length criteria";
+                break;
+            }
+            get_vht_oper(data, results);
+        } break;
+
+        default: {
+            LOG(ERROR) << "Unhandled element received: " << (int)key;
+        } break;
+        }
+
+        ielen -= length + 2;
+        ie += length + 2;
+    }
+}
+
+static bool translate_nl_data_to_bwl_results(sChannelScanResults &results,
+                                             const struct nlattr **bss)
+{
+    //get bssid
+    std::copy_n(results.bssid.oct, sizeof(results.bssid.oct),
+                (unsigned char *)nla_data(bss[NL80211_BSS_BSSID]));
+
+    //get channel and operating frequency band
+    if (bss[NL80211_BSS_FREQUENCY]) {
+        int freq = nla_get_u32(bss[NL80211_BSS_FREQUENCY]);
+        if (freq >= 5180) {
+            results.operating_frequency_band = eOperating_Freq_Band_5GHz;
+        } else {
+            results.operating_frequency_band = eOperating_Freq_Band_2_4GHz;
+        }
+        results.channel = beerocks::utils::wifi_freq_to_channel(freq);
+    }
+
+    // get beacon period
+    if (bss[NL80211_BSS_BEACON_INTERVAL]) {
+        results.beacon_period_ms = (unsigned int)nla_get_u16(bss[NL80211_BSS_BEACON_INTERVAL]);
+    }
+
+    // get signal strength
+    if (bss[NL80211_BSS_SIGNAL_UNSPEC]) {
+        results.signal_strength_dBm = (nla_get_u8(bss[NL80211_BSS_SIGNAL_UNSPEC])) / 100;
+    } else if (bss[NL80211_BSS_SIGNAL_MBM]) {
+        results.signal_strength_dBm = (nla_get_u32(bss[NL80211_BSS_SIGNAL_MBM])) / 100;
+    }
```

Comment:
What if both attributes are missing? Or is it optional?
Shouldn't happen, but still. Maybe log an error? Clear the `signal_strength_dBm` var?

Reply:
same here , we don't have definition for failure.